sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/implementation/KeyClientImpl.java

```diff
@@ -0,0 +1,1420 @@
+
+        return service.updateKeyAsync(vaultUrl, keyProperties.getName(), keyProperties.getVersion(),
+                keyServiceVersion.getVersion(), ACCEPT_LANGUAGE, parameters, CONTENT_TYPE_HEADER_VALUE,
+                context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE))
+            .doOnRequest(ignored -> logger.verbose("Updating key - {}", keyProperties.getName()))
+            .doOnSuccess(response -> logger.verbose("Updated key - {}", response.getValue().getName()))
+            .doOnError(error -> logger.warning("Failed to update key - {}", keyProperties.getName(), error));
+    }
+
+    public Response<KeyVaultKey> updateKeyPropertiesWithResponse(KeyProperties keyProperties, Context context,
+                                                                       KeyOperation... keyOperations) {
+        context = context == null ? Context.NONE : context;
+        KeyRequestParameters parameters = validateAndCreateUpdateKeyRequestParameters(keyProperties, keyOperations);
+        context = enableSyncRestProxy(context);
+
+        return service.updateKey(vaultUrl, keyProperties.getName(), keyProperties.getVersion(),
+                keyServiceVersion.getVersion(), ACCEPT_LANGUAGE, parameters, CONTENT_TYPE_HEADER_VALUE,
+                context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE));
+    }
+
+    private KeyRequestParameters validateAndCreateUpdateKeyRequestParameters(KeyProperties keyProperties,
+                                                                       KeyOperation... keyOperations) {
+        Objects.requireNonNull(keyProperties, "The key properties input parameter cannot be null.");
+        KeyRequestParameters parameters = new KeyRequestParameters()
+            .setTags(keyProperties.getTags())
+            .setKeyAttributes(new KeyRequestAttributes(keyProperties))
+            .setReleasePolicy(keyProperties.getReleasePolicy());
+
+        if (keyOperations.length > 0) {
+            parameters.setKeyOps(Arrays.asList(keyOperations));
+        }
+        return parameters;
+    }
+
+    public PollerFlux<DeletedKey, Void> beginDeleteKeyAsync(String name) {
+        return new PollerFlux<>(getDefaultPollingInterval(),
+            activationOperation(name),
+            createPollOperation(name),
+            (context, firstResponse) -> Mono.empty(),
+            (context) -> Mono.empty());
+    }
+
+    private Function<PollingContext<DeletedKey>, Mono<DeletedKey>> activationOperation(String name) {
+        return (pollingContext) -> withContext(context -> deleteKeyWithResponse(name, context))
+            .flatMap(deletedKeyResponse -> Mono.just(deletedKeyResponse.getValue()));
+    }
+
+    /*
+     * Polling operation to poll on create delete key operation status.
+     */
+    private Function<PollingContext<DeletedKey>, Mono<PollResponse<DeletedKey>>> createPollOperation(String keyName) {
```

Comment:
```suggestion
    private Function<PollingContext<DeletedKey>, Mono<PollResponse<DeletedKey>>> createPollOperationAsync(String keyName) {
```

and some other methods below.

Reply:
This will be handled when Sync Poller work is done.